tests/test_chat.py

```diff
@@ -47,6 +50,72 @@ def multinomial(*_, **__):
         assert torch.cat(actual).tolist() == expected
 
 
+@pytest.mark.parametrize("tokenizer_backend", ["huggingface", "sentencepiece"])
+def test_decode(tokenizer_backend):
+    from lightning.fabric import Fabric
+
+    import chat.base as chat
+
+    class Tokenizer:
+        backend = tokenizer_backend
+        id2token = {1: "foo ", 2: "bar ", 3: "baz "}
+
+        def decode(self, tensor: torch.Tensor) -> str:
+            tensor = [tensor] if tensor.ndim == 0 else tensor
+            return "".join(self.id2token[int(value)] for value in tensor)
+
+    tokenizer_mock = Tokenizer()
+
+    fabric = Fabric()
+    fabric.launch()
```

Comment:
Just in case this was run in a multi-gpu machime

```suggestion
    fabric = Fabric(devices=1, accelerator="cpu")
```

Reply:
Good idea.
Thanks!